### ISSUE TITLE
position not restored when open file from finder on Mac

### ISSUE THREAD

**Issue by khejing** [tmunro/remember-file-positions#10]
Thanks for this great plugin! It always works when I open file from tree view. But when I open file from finder, it doesn't work, and it's the same when open file from desktop shortcut

**Comment by khejing** [comment]
I did some test, found the scroll position will be set as left 0 and top 0.

remember-file-positions.coffee:52 cursor change Point {row: 6, column: 333}
remember-file-positions.coffee:23 file open Point {row: 6, column: 333} Object {top: 0, left: 0}
remember-file-positions.coffee:52 cursor change Point {row: 6, column: 333}
remember-file-positions.coffee:43 set cursor position Point {row: 6, column: 333}
remember-file-positions.coffee:59 scroll change Object {top: 0, left: 0}
remember-file-positions.coffee:52 cursor change Point {row: 6, column: 333}
remember-file-positions.coffee:52 cursor change Point {row: 0, column: 0}

set cursor position worked, but there is something scroll the screen to the top.

**Comment by tmunro** [comment]
Hmm, it looks like this same problem occurs when opening from the File > Open... menu as well. I haven't quite figured out why yet, and following the callstack isn't helping. Currently thinking that I might just ignore `0:0` coordinates or defer my scroll updates until later.

**Comment by tmunro** [comment]
Yeah, I'm at a loss here. I don't understand why there's a difference between opening a file from tree view or through Atom's native opens, as both use `atom.workspace.open`.

**Comment by tmunro** [comment]
Okay, I think I've found the problem. Not sure how I'm going to fix it though, as I think it's a bug in Atom's workspace's `Open` method: https://github.com/atom/atom/blob/v1.17.2/src/workspace.js#L984-L988
```
if (initialLine >= 0 || initialColumn >= 0) {
      if (typeof item.setCursorBufferPosition === 'function') {
        item.setCursorBufferPosition([initialLine, initialColumn])
      }
    }
```
If `options.initialLine` is provided with `null`, it will evaluate `initialLine >= 0` to `true`. But if `options.initialLine` is omitted from the options object, then it will evaluate to false. This is the difference between opening from the tree view and opening from the menu, and I assume from the command line as well.

It shouldn't be too hard a fix, but I'm not set up to write the PR right now. Maybe someone else wants to give it a try?
